sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -127,10 +136,53 @@ def main():
         sbom = json.load(f)
 
     base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
-    if "formulation" in sbom:
-        sbom["formulation"].append({"components": base_images_sbom_components})
+    if args.sbom_type == "cyclonedx":
+        if "formulation" in sbom:
+            sbom["formulation"].append({"components": base_images_sbom_components})
+        else:
+            sbom.update({"formulation": [{"components": base_images_sbom_components}]})
     else:
-        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+        packages = []
+        relationships = []
+        for component in base_images_sbom_components:
+            SPDXID = (
+                f"SPDXRef-{component['type']}-{component['name']}-"
+                + f"{hashlib.sha256(component['purl'].encode()).hexdigest()}"
+            )
+            packages.append(
+                {
+                    "SPDXID": SPDXID,
+                    "name": component["name"],
+                    "externalRefs": [
+                        {
+                            "referenceCategory": "PACKAGE-MANAGER",
+                            "referenceType": "purl",
+                            "referenceLocator": component["purl"],
+                        }
+                    ],
+                    "annotations": [
+                        {
+                            "annotator": "konflux",
+                            "annotationDate": datetime.datetime.now().isoformat(),
+                            "annotationType": "OTHER",
+                            "comment": json.dumps(
+                                {"name": property["name"], "value": property["value"]},
+                                separators=(",", ":"),
+                            ),
+                        }
+                        for property in component["properties"]
+                    ],
+                }
+            )
+            relationships.append(
+                {
+                    "spdxElementId": sbom["SPDXID"],
+                    "relatedSpdxElement": SPDXID,
+                    "relationshipType": "BUILD_TOOL_OF",
```

Comment:
What is `BUILD_TOOL_OF` supposed to represent? A tool used to build the element referenced by `spdxElementId`, i.e. SBOM itself? That would not be the right relationship. The builder images play no part in generating the SBOM.

Reply:
Then I think we're not using it correctly. `SPDXID` refers to a base image, `sbom["SPDXID"]` refers to the SBOM. The base image is not used to build the SBOM.

Reply:
You changed this in a later commit, but I don't understand the change. Could you explain which two elements we are now marking as related?

Reply:
(also, please update these threads when you make relevant changes :pray: )

Reply:
I added extra comments to the code to explain relationship generation